### PR TITLE
add flag for disabling AJAX feature

### PR DESCRIPTION
**Pull request by metal-messiah**
### Overview
This PR adds a flag to enable/disable `All AJAX` feature set.

### CODE REVIEW

File: feature/xhr/aggregate/index.js [hunk 1]
```diff
@@ -98,7 +98,7 @@ function storeXhr(params, metrics, startTime, endTime, type) {
     spaAjaxEvents[interactionId] = spaAjaxEvents[interactionId] || []
     spaAjaxEvents[interactionId].push(event)
   } else {
-    ajaxEvents.push(event)
+    if (allAjaxIsEnabled()) ajaxEvents.push(event)
```
Comment (martinkuba) [inline feature/xhr/aggregate/index.js:101]: I would recommend moving the `allAjaxIsEnabled` call to line 78. That way events will not be created at all.

File: feature/xhr/aggregate/index.js [hunk 2]
```diff
@@ -121,7 +121,7 @@ baseEE.on('interactionDiscarded', function (interaction) {
 function prepareHarvest(options) {
   options = options || {}
 
-  if (ajaxEvents.length === 0) {
+  if (ajaxEvents.length === 0 || !allAjaxIsEnabled()) {
```
Comment (martinkuba) [inline feature/xhr/aggregate/index.js:124]: Instead of checking here, I would recommend not starting the harvesting scheduler at all, which is on line 43 - 44.

File: feature/xhr/aggregate/index.js [hunk 3]
```diff
@@ -233,3 +233,7 @@ function Chunk (events) {
     return this.payload.length * 2 > maxPayloadSize
   }
 }
+
+function allAjaxIsEnabled() {
+  return (!!config.getConfiguration('ajax.enabled') || false)
```
Comment (martinkuba) [inline feature/xhr/aggregate/index.js:238]: I don't think you need the `|| false` the way it is written.

**Review by martinkuba — APPROVED** [review]
